### DISCUSSION TITLE
How to replace BN with syncBN in a  model(pytorch version)?

### DISCUSSION THREAD

**Discussion by shufangxun** [horovod/horovod#2780] [Q&A]
## code
![image](https://user-images.githubusercontent.com/19266420/106712394-d8b46b80-6633-11eb-9121-bb684dfe749b.png)

## problem 
TypeError: horovod.torch.sync_batch_norm.SyncBatchNorm is not a Module subclass

**Comment by shufangxun** [comment]
@romerojosh plz

**Comment by tgaddair** [comment]
Hey @shufangxun, can you share a minimal repro script?  Also, what versions of PyTorch and Horovod are you using?

**Comment by shufangxun** [comment]
hello,  i have fixed it with apex.parallel.convert_syncbn_model. By the way，does horovod support ReduceOp like torch.distributed.ReduceOp，which includes MAX and MIN

**Comment by tgaddair** [comment]
Hey @shufangxun, glad you got it working!

Currently supported ops for allreduce are sum and average.  We can certainly add max and min as well.  Is this something you need for your project?

**Comment by shufangxun** [comment]
yes, i need to get max value of the outputs after softmax within all GPUs. I have changed to pytorch ddp.
